opencog/atoms/pattern/PatternUtils.cc

```diff
@@ -22,14 +22,17 @@
  */
 
 #include <opencog/atomutils/FindUtils.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <boost/range/algorithm/find.hpp>
```

Comment:
Gahh, can we avoid using boost, if possible? Its a bunch of nasty, poorly designed code, I'm tired of having to debug the issues that it causes.

Reply:
We can, but so you know boost.range is extremely mature, I'm been using it for years, never had a problem once (unlike boost.graph for instance). BTW, range is coming to the standard https://en.wikipedia.org/wiki/C%2B%2B20 BTW but it's still 3 to 4 years away.